src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -282,18 +285,18 @@ public void deleteBuildLogEntriesForProgrammingSubmission(ProgrammingSubmission
      */
     public void saveBuildLogsToFile(List<BuildLogEntry> buildLogEntries, String resultId) {
 
-        Path buildLogsPath = Path.of("buildLogs");
+        Path buildLogs = Path.of(buildLogsPath);
 
-        if (!Files.exists(buildLogsPath)) {
+        if (!Files.exists(buildLogs)) {
             try {
-                Files.createDirectory(buildLogsPath);
+                Files.createDirectory(buildLogs);
```

Comment:
```suggestion
                Files.createDirectories(buildLogs);
```
The operation would fail otherwise in case the build log path is specified as something as `./build/logs/in/subdir/` and only `./build/logs/` exists at the moment.